packages/slice-machine/src/tracker.ts

```diff
@@ -0,0 +1,176 @@
+import type { Analytics as ClientAnalytics } from "@segment/analytics-next";
+import { AnalyticsBrowser } from "@segment/analytics-next";
+import { Frameworks } from "@slicemachine/core/build/src/models";
+import { LibraryUI } from "@models/common/LibraryUI";
+
+// These events should be sync with the tracking Plan on segment.
+type AllSliceMachineEventType = EventType | ContinueOnboardingType;
+
+enum EventType {
+  Review = "SliceMachine Review",
+  OnboardingStart = "SliceMachine Onboarding Start",
+  OnboardingSkip = "SliceMachine Onboarding Skip",
+  SlicePreviewSetup = "SliceMachine Slice Preview Setup",
+  SlicePreviewOpen = "SliceMachine Slice Preview Open",
+}
+
+export enum ContinueOnboardingType {
+  OnboardingContinueIntro = "SliceMachine Onboarding Continue Screen Intro",
+  OnboardingContinueScreen1 = "SliceMachine Onboarding Continue Screen 1",
+  OnboardingContinueScreen2 = "SliceMachine Onboarding Continue Screen 2",
+  OnboardingContinueScreen3 = "SliceMachine Onboarding Continue Screen 3",
+}
+
+let _client: ClientAnalytics | null = null;
+let _isTrackingActive = true;
+let _repoName: string | null = null;
```

Comment:
same for this, it should be contained in the closure. But ideally it should be a class, if you have all these variables it means that the tracker is stateful here.

Reply:
Done